MIGRATING_v2_to_v3.md

```diff
@@ -0,0 +1,223 @@
+# Okta Java Management SDK Migration Guide
+ 
+This SDK uses semantic versioning and follows Okta's 
+[library version policy](https://developer.okta.com/code/library-versions/). 
+In short, we do not make breaking changes unless the major version changes!
+
+## Migrating from 2.x.x to 3.0.0
+
+Version 3.0.0 of this SDK introduces a number of breaking changes from previous versions. 
+In addition to many new classes/interfaces, some existing classes/interfaces are no longer backward compatible due to method renaming and signature changes.
+
+### Package `com.okta.sdk.client.Client`
+
+Below method has been added.
+   - `void deleteIdentityProviderKey(String keyId)`
+   
+Below methods have undergone a signature change.
+   - `listGroups(String q, String filter)` signature changed to `listGroups(String q, String filter, String expand)`
```

Comment:
Please add a note that `expand` is OPTIONAL (https://developer.okta.com/docs/reference/api/groups/#request-parameters-3).

Reply:
updated